src/main/java/DukeController.java

```diff
@@ -0,0 +1,80 @@
+import java.util.Scanner;
+
+public class DukeController {
+
+    public static final int EVENT_STRING_LENGTH = 5;
+    public static final int DEADLINE_STRING_LENGTH = 8;
+    public static final int TODO_STRING_LENGTH = 4;
+    public static Task[] tasks = new Task[100];
+    public static int taskCount = 0;
+
+    public static void run(Scanner scanner) {
+        while (true) {
+            String input = scanner.nextLine();
+            if (input.equals("bye")) {
+                return;
+            } else if (input.equals("list")) {
+                UI.printTaskList(tasks, taskCount);
+            } else if (input.contains("done")) {
+                markTaskDone(input);
+            } else {
+                addNewTask(input);
+            }
+        }
+    }
+
+    public static void markTaskDone(String input) {
+        String[] parts = input.split(" ");
+        int taskIndex = Integer.parseInt(parts[1]) - 1;
+        Task taskToMarkDone = tasks[taskIndex];
+        taskToMarkDone.markDone();
+        UI.taskMarkedAsDoneMessage(taskToMarkDone);
+    }
+
+    public static void addNewTask(String input) {
+        Task taskToAdd = processTaskToAdd(input);
+        if (taskToAdd != null) {
```

Comment:
not sure but java coding standards mentioned something on the .equals. Not too sure if this applies to your case.